src/DynamoPackagesUI/Utilities/PackageManagerCommands.cs

```diff
@@ -0,0 +1,479 @@
+
+
+        private dynamic _downloadRequest;
+        public dynamic DownloadRequest
+        {
+            get { return _downloadRequest; }
+            set
+            {
+                if (value is Newtonsoft.Json.Linq.JObject)
+                    _downloadRequest = value;
+                else
+                    _downloadRequest = JsonConvert.DeserializeObject<dynamic>(value.ToString());
+            }
+        }
+
+        private dynamic _pkgRequest;
+        public dynamic PkgRequest
+        {
+            get { return _pkgRequest; }
+            set
+            {
+                if (value is Newtonsoft.Json.Linq.JObject)
+                    _pkgRequest = value;
+                else
+                    _pkgRequest = JsonConvert.DeserializeObject<dynamic>(value);
+            }
+        }
+
+        private string PackageInstallPath { get; set; }
+
+        /// <summary>
+        /// Install Dynamo Package
+        /// </summary>
+        /// <param name="downloadPath"></param>
+        private void InstallPackage(string downloadPath)
+        {
+            var firstOrDefault = Loader.LocalPackages.FirstOrDefault(pkg => pkg.ID == DownloadRequest.asset_id.ToString());
+            if (firstOrDefault != null)
+            {
+                var dynModel = Model;
+                try
+                {
+                    firstOrDefault.UninstallCore(dynModel.CustomNodeManager, Loader, dynModel.PreferenceSettings);
+                }
+                catch
+                {
+                    MessageBox.Show(String.Format(Resources.MessageFailToUninstallPackage,
+                        ResourceProvider.ProductName,
+                        DownloadRequest.asset_name.ToString()),
+                        Resources.UninstallFailureMessageBoxTitle,
+                        MessageBoxButton.OK, MessageBoxImage.Error);
+                }
+            }
+
+            var settings = Model.PreferenceSettings;
+            PackageDownloadHandle packageDownloadHandle = new PackageDownloadHandle();
+            packageDownloadHandle.Name = DownloadRequest.asset_name;
+            packageDownloadHandle.Done(downloadPath);
+
+            //string installedPkgPath = string.Empty;
+            Package dynPkg = null;
+            if (packageDownloadHandle.Extract(Model, this.PackageInstallPath, out dynPkg))
+            {
+                var p = Package.FromDirectory(dynPkg.RootDirectory, Model.Logger);
+                p.ID = DownloadRequest.asset_id;
+                Application.Current.Dispatcher.Invoke((Action)(() =>
+                {
+                    Loader.Load(p);
+                }));
+                packageDownloadHandle.DownloadState = PackageDownloadHandle.State.Installed;
+                this.PackageInstallPath = string.Empty;
+            }
+        }
+
+        /// <summary>
+        /// Download and Install Dynamo Package
+        /// </summary>
+        /// <param name="pkg"></param>
```

Comment:
Can you please document the format of pkg, it seems it's a comma separated string. What does each part consist of?